src/sys/windows.rs

```diff
@@ -774,6 +775,53 @@ pub(crate) fn to_mreqn(
     }
 }
 
+#[allow(unused_unsafe)] // TODO: replace with `unsafe_op_in_unsafe_fn` once stable.
+pub(crate) fn unix_sockaddr(path: &Path) -> io::Result<SockAddr> {
+    unsafe {
+        SockAddr::try_init(|storage, len| {
+            // Safety: `SockAddr::init` zeros the address, which is a valid
+            // representation.
+            let storage: &mut windows_sys::Win32::Networking::WinSock::sockaddr_un =
+                unsafe { &mut *storage.cast() };
+            let len: &mut socklen_t = unsafe { &mut *len };
+
+            // Windows expects a UTF-8 path here even though Windows paths are
+            // usually UCS-2 encoded. If Rust exposed OsStr's Wtf8 encoded
+            // buffer, this could be used directly, relying on Windows to
+            // validate the path, but Rust hides this implementation detail.
+            let bytes = path
+                .to_str()
+                .ok_or_else(|| {
+                    io::Error::new(io::ErrorKind::InvalidInput, "path must be valid UTF-8")
+                })?
+                .as_bytes();
+
+            // Windows appears to allow non-null-terminated paths, but this is
```

Comment:
I don't remember where I read this, but the a person that was part of the team that wrote this code said it was an oversight in the docs and they would make an internal issue to update the docs to guarantee this. So, we should probably keep the code as is for the moment and open an issue to allow it in the future (in case some really needs that one additional byte).

Reply:
Right, that was my intention (I'm also the one who wrote the comment about it being an oversight in the documentation). I can open an issue and reference it here.

Reply:
#331.